### PR TITLE
fix: using new "--data-home" config for Datanode

### PR DESCRIPTION
**Pull request by MichaelScofield**

### CODE REVIEW

File: pkg/deployer/baremetal/deployer.go [hunk 1]
```diff
@@ -173,9 +175,62 @@ func (d *Deployer) CreateEtcdCluster(ctx context.Context, clusterName string, op
...
+	exists, err := utils.FileExists(etcdctlBin)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		d.logger.V(3).Infof("'etcdctl' is not found under the same directory of 'etcd', skip checking the healthy of Etcd.")
+		return nil
+	}
+
+	for retry := 0; retry < 10; retry++ {
```
Comment (zyy17) [inline pkg/deployer/baremetal/deployer.go:197]: question: Can we use `etcdctl --endpoints=$ENDPOINTS endpoint health`([link](https://etcd.io/docs/v3.5/tutorials/how-to-check-cluster-status/))? It looks simpler.

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -24,3 +24,15 @@ func CreateDirIfNotExists(dir string) (err error) {
 	}
 	return nil
 }
+
+func FileExists(filepath string) (bool, error) {
```
Comment (zyy17) [inline pkg/utils/utils.go]: Suggestion: The following functon is more Go way:

```go
func IsFileExists(filepath string) (bool, error) {
	info, err := os.Stat(filepath)
	if os.IsNotExist(err) {
		// file does not exist
		return false, nil
	}

	if err != nil {
		// Other errors happened.
		return false, err
	}

	if info.IsDir() {
		// It's a directory.
		return false, fmt.Errorf("'%s' is directory, not file", filepath)
	}

	// The file exists.
	return true, nil
}
```

File: pkg/deployer/baremetal/deployer.go (hunk 1 above)
Comment (MichaelScofield) [reply pkg/deployer/baremetal/deployer.go:197]: We need to make sure the Etcd do have a leader. I don't know if `health` check can be sure about that. Seems there are no "leader" info in it.

**Comment by MichaelScofield** [comment]
@zyy17 PTAL

**Review by zyy17 — APPROVED** [review]
LGTM

**Pull request merged**